### ISSUE TITLE
People Counter

### ISSUE THREAD

**Issue by laercionit** [xibosignage/xibo#50]
People Counter

I believe this is a trend in the market, integrate the system to a counter of people who went and looked at the display.

We could have a WebCam, connected to android. The software android would have the option to enable Track People. If enabled, select the webcam.

This information is valuable, could be presented in Statistics, part of the Layout or Display.

Not complex but has a potential for the market.

I hope you like the idea DAN.

**Comment by dasgarner** [comment]
Applies to all players I think.

**Comment by knitehawk** [comment]
you could also just connect it to a motion sensor that would then activate the camera, or send a signal (Serial, or other) to counter

**Comment by laercionit** [comment]
Ideally recognize the USB camera compatible with the player (Android, WIndows, Linux). Then apply a face recognizer. There bilbiotecas ready for it. Thus not only accounts for the people who have to screen, keep a record of all who were looking at the screen.

This information would be sent to the CMS usually presenting in statistics and content

**Comment by dasgarner** [comment]
My feeling is that this involves some fairly complicated face recognition technology (gender detection, eye tracking, etc) and that an interface to a 3rd party solution would be the most likely route (or indeed a library project - if there is such a thing).

**Comment by laercionit** [comment]
Dan, There are libraries for Java and AspNet ready to smile and face detect

**Comment by dasgarner** [comment]
OK great - it would be good to have the links in here for reference. Depending on how the integration was managed they may/may not need to be open source.

**Comment by laercionit** [comment]
Java OpenCV Bindings

https://www.youtube.com/watch?v=7bpwJtQ65oY
https://www.youtube.com/watch?v=gcONLfkFSNM
http://answers.opencv.org/question/12978/face-detection-through-a-webcam-in-java/

**Comment by nagyrobi** [comment]
Uh Oh just a chime in here, in many places around the world placement of webcams like this rises privacy issues. Displays in public areas (public means really public: open streets, parks etc) may have such cameras, but displays in private areas (like shops, malls, gas stations, cinemas) can only have cameras if there's an autorization for that.
Thus in places like this people counter could be fesable only with motion detectors, which would just simply count the motion occurences.

**Comment by dasgarner** [comment]
I think that only counting motion defeats the object slightly as the data is not that useful. If there are privacy concerns then I think that is up to the signage operator to manage and enable/disable any people counting functionality. In much the same way as it is their responsibility to ensure people are not putting copyright video on their signs.

I did find this (http://opencv.org/) which might provide an open source solution that can be used cross platform. There are also a bunch of commercial ones we could integrate with.

**Comment by nagyrobi** [comment]
Agree, my chime was in order to keep in mind that this could conflict with legislation in some countries, so software should be developed in a manner that this feature should be possible to be completely uninstalled, or maybe installed optionally as a plugin or a package or something similar. Legislation specifies that in certain areas "the installation of equipment able to do this is prohibited", so that means, the deployer of the signage has to be able to prove that the screen hardware under no circumstances offers any possibility to do that.
Otherwise Xibo can be de-installed from these locations and something else must be put instead... which is not desirable for the project, I guess...

**Comment by dasgarner** [comment]
Interesting point, so this legislation would force an entirely separate piece of software that can be completely removed (i.e. not installed). Which countries are we talking about here?

Most of the "people counter" software I have seen acts as a integral part of the host application (integrated with SDK) - the thought of maintaining two version of the player doesn't appeal to me that much :smile:

**Comment by nagyrobi** [comment]
Mainly Easten Europe.

**Comment by laercionit** [comment]
Dan
We're talking about allowing the APP recognize a webcam, select the device and collect data.

For countries where this feature is not allowed is not the responsibility system, but the user does not use. The system is global, each location if you need to be aware that restricting the use or not of the resource.

**Comment by dasgarner** [comment]
We aren't lawyers so we don't really know - it seems to me that installing a device without a webcam would solve the issue completely (even if the software provides the functionality, if the hardware isn't there - it cannot be used).

Happy with that @nagyrobi ?

**Comment by nagyrobi** [comment]
Happy as long as it's possible to completely disable the functionality in the software too. Things like
- there should be no sign in the client's interface that it has this functionality if the feature is disabled
- there should be a global switch in the CMS to enable or disable this entire functionality, disabling it would hide these options not only in the client but in the CMS too - I'd think a value in the editable config file would be fine
- and vice-versa: with this feature enabled, both the client and the CMS should make well visible that the product has such features
The idea behind is: the authority delegate when looks at the system, should be able to observer and understand clearly and simply, just by looking at it, that it has or it has not the features in case.

**Comment by supportreq** [comment]
Really important are we implementing this??

**Comment by dasgarner** [comment]
You can see further discussion on this here: https://community.xibo.org.uk/t/number-of-people-who-viewed-a-specific-screen/721
